packages/near-api-js/src/transaction_builder.ts

```diff
@@ -0,0 +1,84 @@
+import BN from 'bn.js';
+import { FunctionCallOptions } from './account';
+import { Connection } from './connection';
+import { DEFAULT_FUNCTION_CALL_GAS } from './constants';
+import { AccessKeyView, FinalExecutionOutcome } from './providers/provider';
+import { AccessKey, Action, addKey, createAccount, deleteAccount, deleteKey, deployContract, fullAccessKey, functionCall, SignedTransaction, stake, transfer } from './transaction';
+import { TransactionSender } from './transaction_sender';
+import { PublicKey } from './utils';
+
+/**
+ * Transaction Builder class. Initialized to an account that will sign the final transaction
+ */
+export class TransactionBuilder extends TransactionSender {
+    readonly receiverId: string;
+    readonly senderId: string;
+    readonly connection: Connection;
+
+    readonly actions: Action[];
+    accessKeyByPublicKeyCache: { [key: string]: AccessKeyView };
+
+    constructor(connection: Connection, senderId: string, receiverId: string) {
+        super(connection, senderId);
+        this.receiverId = receiverId;
+        this.connection = connection;
+        this.senderId = senderId;
+        this.actions = [];
+        this.accessKeyByPublicKeyCache = {};
+    }
+
+    addKey(publicKey: string | PublicKey, accessKey: AccessKey = fullAccessKey()): this {
+        this.actions.push(addKey(PublicKey.from(publicKey), accessKey));
```

Comment:
Any reason not to use the exposed method rather than mutating the state directly? Using `this.push` here instead ensures there's only one way that state gets mutated, and makes the code less brittle.